extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/supports/BlockingServerInterceptor.java

```diff
@@ -96,11 +97,17 @@ synchronized void setDelegate(ServerCall.Listener<ReqT> delegate) {
 
         private synchronized void executeOnContextOrEnqueue(Consumer<ServerCall.Listener<ReqT>> consumer) {
             if (this.delegate != null) {
+                final Context grpcContext = Context.current();
                 vertx.executeBlocking(new Handler<Promise<Object>>() {
                     @Override
                     public void handle(Promise<Object> f) {
+                        final Context previous = Context.current();
+                        grpcContext.attach();
+
                         consumer.accept(delegate);
                         f.complete();
+
+                        grpcContext.detach(previous);
```

Comment:
@cescoffier maybe it's not mandatory in this case but we usually restore previous contexts in a finally block. It's not needed here?

Reply:
I think it's best to keep it. In my opinion this part of the code is responsible to wrap the context. And it should not rely on any other code that will eventually drop the context. This could have sever security implication. I use the context to store the JWT token. If the context is not cleared correctly and somehow the jwt end up in another request it's a security leak.

Reply:
Maybe I wasn't clear.

My question was: shouldn't this code look like:
```java
final Context previous = Context.current();
grpcContext.attach();
try {
    consumer.accept(delegate);
    f.complete();
} finally {
    grpcContext.detach(previous);
}
```

Reply:
(you might want to put the `attach` in the `try` too, no idea what it does :))

Reply:
Hoo my bad, then I agree, we should do the detach in the finally.

Not sure about the attach. But in the javadoc example it's outside the try: https://grpc.github.io/grpc-java/javadoc/io/grpc/Context.html#attach--

Reply:
@gsmet done